components/gantry/oneaxis/oneaxis.go

```diff
@@ -172,67 +172,50 @@ func (g *oneAxis) home(ctx context.Context, np int) error {
-	// zero position of the oneaxis, and adds a second position limit based on the steps per length.
-	case 1:
-		if err := g.homeOneLimSwitch(ctx); err != nil {
-			return err
-		}
-	// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
-	// zero position of the oneaxis, then go till it hits the second limit switch, then encode that position as the
-	// at-length position of the oneaxis.
-	case 2:
-		if err := g.homeTwoLimSwitch(ctx); err != nil {
+
+	if np > 0 {
+		// An axis with one limit switch will go till it hits the limit switch, encode that position as the
+		// zero position of the oneaxis, and adds a second position limit based on the steps per length.
+		// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
+		// zero position of the oneaxis, then go till it hits the second limit switch, then encode that position as the
+		// at-length position of the oneaxis.
+		if err := g.homeLimSwitch(ctx); err != nil {
 			return err
 		}
-	// An axis with an encoder will encode the
-	case 0:
+	} else {
+		// An axis with an encoder will encode the
 		if err := g.homeEncoder(ctx); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (g *oneAxis) homeTwoLimSwitch(ctx context.Context) error {
+func (g *oneAxis) homeLimSwitch(ctx context.Context) error {
 	positionA, err := g.testLimit(ctx, true)
 	if err != nil {
 		return err
 	}
 
-	positionB, err := g.testLimit(ctx, false)
-	if err != nil {
-		return err
+	positionB := 0.0
+	if len(g.limitSwitchPins) > 1 {
+		// Multiple limit switches, get positionB from testLimit
+		positionB, err = g.testLimit(ctx, false)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Only one limit switch, calculate positionB
+		revPerLength := g.lengthMm / g.mmPerRevolution
+		positionB = positionA + revPerLength
 	}
 
 	g.positionLimits = []float64{positionA, positionB}
 	g.positionRange = positionB - positionA
+	g.logger.Debugf("positionA: %0.2f positionB: %0.2f range: %0.2f", g.positionLimits[0], g.positionLimits[1], g.positionRange)
 
-	g.logger.Infof("positionA: %0.2f positionB: %0.2f range: %0.2f", g.positionLimits[0], g.positionLimits[1], g.positionRange)
-
-	// Go backwards so limit stops are not hit.
-	x := g.gantryToMotorPosition(0.8 * g.lengthMm)
-	if err = g.motor.GoTo(ctx, g.rpm, x, nil); err != nil {
-		return err
-	}
-	return nil
-}
-
-func (g *oneAxis) homeOneLimSwitch(ctx context.Context) error {
-	// One pin always and only should go backwards.
-	positionA, err := g.testLimit(ctx, true)
-	if err != nil {
-		return err
-	}
-
-	revPerLength := g.lengthMm / g.mmPerRevolution
-	positionB := positionA + revPerLength
-
-	g.positionLimits = []float64{positionA, positionB}
-
-	// Go backwards so limit stops are not hit.
-	x := g.gantryToMotorPosition(0.2 * g.lengthMm)
+	// Go to middle so limit stops are not hit.
+	x := g.gantryToMotorPosition(0.5 * g.lengthMm)
```

Comment:
I'd like to move to 0.2*glengthMm for one limit switch and 0.8*g.lengthMm for the single limit switch case, so we're not waiting too long after the homing sequence.